lib/ipfs/IpfsStorage.ts

```diff
@@ -45,32 +61,24 @@ export default class IpfsStorage {
     // If we hit error attempting to fetch the content metadata, return not-found.
     let contentMetadata = undefined;
     try {
-      contentMetadata = await (this.node as any).object.stat(hash);
+      contentMetadata = await this.node.object.stat(hash);
     } catch (error) {
-      console.info(error);
+      console.debug(`Error thrown while executing ipfs.stat for CID ${hash}: ${JSON.stringify(error, Object.getOwnPropertyNames(error))}`);
       return { code: FetchResultCode.NotFound };
     }
 
-    // If content size cannot be found, return not-found.
-    if (contentMetadata === undefined || contentMetadata.CumulativeSize === undefined) {
+    if (contentMetadata === undefined || contentMetadata.DataSize === undefined) {
       return { code: FetchResultCode.NotFound };
     }
 
-    // NOTE: IPFS API does not have an API for finding out only the content size,
-    // IPFS API only provides the "cumulative size" that includes the additional metadata on how the Merkle DAG is formed.
-    // So here we account for the additional space used by performing a more lenient max size check by 10%,
-    // we will track the exact conent size of content later when we fetch the content if the content passes this size check.
-    const adjustedMaxSize = maxSizeInBytes * 1.1;
-    if (contentMetadata.CumulativeSize > adjustedMaxSize) {
-      console.info(`Cumulative size of ${contentMetadata.CumulativeSize} bytes is greater than the ${adjustedMaxSize} cumulative size limit.`);
+    if (contentMetadata.DataSize > maxSizeInBytes) {
+      console.info(`Size of ${contentMetadata.DataSize} bytes is greater than the ${maxSizeInBytes} data size limit for CID ${hash}.`);
```

Comment:
I think this can cause legitimate files to be wrongly considered as bad. Let's file a bug to optimize this part, I suspect that we will still need to use the same/similar cumulative size check for optimization.

Reply:
Will remove the check completely, and create an optimization task.